.github/workflows/aws_deploy.yml

```diff
@@ -59,13 +59,17 @@ jobs:
         run: |
           echo "image-exists=$(if aws ecr list-images --repository-name=$repository --query "imageIds[*].imageTag" | grep -q ${{ github.sha }}; then echo true; else echo false; fi)" >> $GITHUB_ENV
 
-      - name: Build, tag, and push docker image to ECR
+      - name: Build, tag, and push docker image to ECR if there is no image, failing for releases
         id: build-image
         if: ${{ env.image-exists == 'false' }}
         env:
           registry: ${{ steps.ecr-login.outputs.registry }}
           commit_tag: ${{ github.sha }}
         run: |
+          if [[ ${{ inputs.environment }} == 'production' ]]; then
```

Comment:
This mostly looks good to me - just wanted to check we're not deleting images from the registry (e.g. due to age), first.

Reply:
Yep I'll look into this, good point 👍

Reply:
Thanks!